File: 2020/238. Product of Array Except Self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] ans = new int[nums.length];
        int left = 1, right = 1;
        for (int i = 0; i < nums.length; i++) {
            ans[i] = left;
            left *= nums[i];
        }
        for (int i = nums.length - 1; i > -1; i--) {
            ans[i] *= right;
            right *= nums[i];
        }
        return ans;
    }
}

File: 2024/238. Product of Array Except Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> res;
        int prod = 1;
        for (int i = 0; i < nums.size(); i++) {
            res.push_back(prod);
            prod *= nums[i];
        }
        prod = 1;
        for (int i = nums.size() - 1; i >= 0; i--) {
            res[i] *= prod;
            prod *= nums[i];
        }
        return res;
    }
};